manifests/config/pki.pp

```diff
@@ -3,17 +3,25 @@
 # Ensures that there are PKI certificates readable by the rsyslog user in
 # /etc/sssd/pki
 #
-class sssd::config::pki {
+class sssd::config::pki
+{
   assert_private()
 
-  unless empty($::sssd::cert_source) { validate_absolute_path($::sssd::cert_source) }
 
-  if $::sssd::use_simp_pki {
-    include '::pki'
-    ::pki::copy { '/etc/sssd': }
+  $pki = $::sssd::pki
+  $source = $::sssd::app_pki_cert_source
+  $app_pki_dir = $::sssd::app_pki_dir
+
```

Comment:
This class should ensure app_pki_dir exists before pki::copy/file{app_pki_dir/pki}